pkg/build/controller/controller.go

```diff
@@ -102,9 +102,13 @@ func (bc *BuildController) HandleBuild(build *buildapi.Build) error {
 	}
 
 	if buildutil.IsBuildComplete(build) {
+		if _, ok := build.Annotations[buildapi.BuildCompletedAnnotation]; ok {
+			return nil
+		}
```

Comment:
@smarterclayton how is building a cache better ? It really doesn't matter that we run the policy.OnComplete again, but we'd rather avoid it if we can.